### PR TITLE
Sm/parking orbit

### PR DESCRIPTION
**Pull request by mshanemc**
> requires https://github.com/forcedotcom/sfdx-core/pull/586 , https://github.com/forcedotcom/source-tracking/pull/164 and  https://github.com/forcedotcom/source-deploy-retrieve/pull/595

### What does this PR do?
primary: parking orbit
secondary features/changes
- source:delete will tell users that it's running "checkonly" if that's specified
- remove -a from source:status examples (flag had been removed, examples still had it)
- minor json output changes: true/false are now boolean (instead of "true"/"false"] and line numbers are numbers!
- source:retrieve without anything specified to retrieve will throw an error 

### What issues does this PR fix or reference?
[@W-10743859@](https://gus.lightning.force.com/a07EE00000rPLYGYA4)

### Testing
you'll need the branches for each of the 3 PRs at the top of this PR
1. link core into SDR
2. link core and SDR into STL
3. link core, SDR, and STL into plugin-source

### Shipping
1. release core
2. bump core in SDR and release
3. bump core and SDR in STL and release
4. bump core, SDR, and STL in plugin-source and release

### CODE REVIEW

File: test/commands/mdapi/retrieve.test.ts [hunk 1]
```diff
@@ -140,12 +143,12 @@ describe('force:mdapi:retrieve', () => {
...
-    expect(lifecycleEmitStub.calledOnce, failureMsg).to.equal(true);
-    expect(lifecycleEmitStub.firstCall.args[0]).to.equal('preretrieve');
-    expect(lifecycleEmitStub.firstCall.args[1]).to.deep.equal({ packageXmlPath: manifest });
-  };
+  // const ensureHookArgs = (manifest?: string) => {
+  //   const failureMsg = 'Lifecycle.emit() should be called for preretrieve';
+  //   expect(lifecycleEmitStub.calledOnce, failureMsg).to.equal(true);
+  //   expect(lifecycleEmitStub.firstCall.args[0]).to.equal('preretrieve');
+  //   expect(lifecycleEmitStub.firstCall.args[1]).to.deep.equal({ packageXmlPath: manifest });
+  // };
```
Comment (shetzel) [inline test/commands/mdapi/retrieve.test.ts:151]: Why are we no longer verifying the hooks?

File: test/commands/source/delete.test.ts [hunk 2]
```diff
@@ -122,19 +126,19 @@ describe('force:source:delete', () => {
...
-    expect(lifecycleEmitStub.calledTwice, failureMsg).to.equal(true);
-    expect(lifecycleEmitStub.firstCall.args[0]).to.equal('predeploy');
-    expect(lifecycleEmitStub.secondCall.args[0]).to.equal('postdeploy');
-  };
+  // const ensureHookArgs = () => {
+  //   const failureMsg = 'Lifecycle.emit() should be called for predeploy and postdeploy';
+  //   expect(lifecycleEmitStub.calledTwice, failureMsg).to.equal(true);
+  //   expect(lifecycleEmitStub.firstCall.args[0]).to.equal('predeploy');
+  //   expect(lifecycleEmitStub.secondCall.args[0]).to.equal('postdeploy');
+  // };
```
Comment (shetzel) [inline test/commands/source/delete.test.ts:134]: Why is this commented now?

File: test/commands/source/deploy.test.ts [hunk 3]
```diff
@@ -170,14 +170,14 @@ describe('force:source:deploy', () => {
...
-    expect(lifecycleEmitStub.secondCall.args[1]).to.deep.equal(deployResult);
-  };
+  // const ensureHookArgs = () => {
+  //   const failureMsg = 'Lifecycle.emit() should be called for predeploy and postdeploy';
+  //   expect(lifecycleEmitStub.calledTwice, failureMsg).to.equal(true);
+  //   expect(lifecycleEmitStub.firstCall.args[0]).to.equal('predeploy');
+  //   expect(lifecycleEmitStub.firstCall.args[1]).to.deep.equal([exampleSourceComponent]);
+  //   expect(lifecycleEmitStub.secondCall.args[0]).to.equal('postdeploy');
+  //   expect(lifecycleEmitStub.secondCall.args[1]).to.deep.equal(deployResult);
+  // };
```
Comment (shetzel) [inline test/commands/source/deploy.test.ts:180]: same hooks comment as above

File: test/commands/source/deploy.test.ts [hunk 4]
```diff
@@ -466,7 +466,7 @@ describe('force:source:deploy', () => {
     });
   });
 
-  describe('Hooks', () => {
+  describe.skip('Hooks', () => {
```
Comment (shetzel) [inline test/commands/source/deploy.test.ts:469]: why skip?

File: test/commands/source/retrieve.test.ts [hunk 5]
```diff
@@ -152,22 +157,22 @@ describe('force:source:retrieve', () => {
...
-    expect(lifecycleEmitStub.secondCall.args[1]).to.deep.equal(expectedResults.inboundFiles);
-  };
+  // const ensureHookArgs = () => {
+  //   const failureMsg = 'Lifecycle.emit() should be called for preretrieve and postretrieve';
+  //   expect(lifecycleEmitStub.calledTwice, failureMsg).to.equal(true);
+  //   expect(lifecycleEmitStub.firstCall.args[0]).to.equal('preretrieve');
+  //   expect(lifecycleEmitStub.firstCall.args[1]).to.deep.equal([exampleSourceComponent]);
+  //   expect(lifecycleEmitStub.secondCall.args[0]).to.equal('postretrieve');
+  //   expect(lifecycleEmitStub.secondCall.args[1]).to.deep.equal(expectedResults.inboundFiles);
+  // };
```
Comment (shetzel) [inline test/commands/source/retrieve.test.ts:167]: same hooks comment

**Review by shetzel — CHANGES_REQUESTED** [review]

File: .circleci/config.yml [hunk 6]
```diff
@@ -1,6 +1,6 @@
 version: 2.1
 orbs:
-  release-management: salesforce/npm-release-management@4
```
Comment (iowillhoit) [inline .circleci/config.yml:3]: @mshanemc why this change?

Comment (mshanemc) [reply .circleci/config.yml:3]: nice catch.  That was a branch that's been merged so it can go back to normal.

**Review by iowillhoit — APPROVED** [review]
